test/capi/capi_sarray.cxx

```diff
@@ -6,42 +6,1581 @@
 #include <vector>
 #include "capi_utils.hpp"
 
+#include <unity/lib/gl_sframe.hpp>
+#include <unity/lib/gl_sarray.hpp>
+#include <capi/impl/capi_sarray.hpp>
+#include <capi/impl/capi_flexible_type.hpp>
+#include <capi/impl/capi_flex_list.hpp>
+#include <iostream>
+
 class capi_test_sarray {
  public:
 
  void test_sarray_double() {
 
     std::vector<double> v = {1, 2, 4.5, 9, 10000000, -12433};
 
-    tc_error* error = NULL; 
-    
-    tc_flex_list* fl = make_flex_list_double(v); 
+    tc_error* error = NULL;
+
+    tc_flex_list* fl = make_flex_list_double(v);
 
     tc_sarray* sa = tc_sarray_create_from_list(fl, &error);
 
-    TS_ASSERT(error == NULL); 
+    TS_ASSERT(error == NULL);
 
-    { 
+    {
       // Make sure it gets out what we want it to.
-      for(size_t i = 0; i < v.size(); ++i) { 
-        tc_flexible_type* ft = tc_sarray_extract_element(sa, i, &error); 
-        TS_ASSERT(error == NULL); 
+      for(size_t i = 0; i < v.size(); ++i) {
+        tc_flexible_type* ft = tc_sarray_extract_element(sa, i, &error);
+        TS_ASSERT(error == NULL);
 
         TS_ASSERT(tc_ft_is_double(ft) != 0);
 
-        double val = tc_ft_double(ft, &error); 
-        TS_ASSERT(error == NULL); 
+        double val = tc_ft_double(ft, &error);
+        TS_ASSERT(error == NULL);
+
+        TS_ASSERT(v[i] == val);
 
-        TS_ASSERT(v[i] == val); 
-      
         tc_ft_destroy(ft);
       }
     }
   }
-};
 
-BOOST_FIXTURE_TEST_SUITE(_capi_test_sarray, capi_test_sarray)
-BOOST_AUTO_TEST_CASE(test_sarray_double) {
-  capi_test_sarray::test_sarray_double();
+  void test_tc_op_sarray_lt_sarray(){
+    tc_error* error = NULL;
+
+    std::vector<double> v1 = {1, 2, 4.5, 9, 389, 23};
+    std::vector<double> v2 = {3, 2, 23, 53, 32, 345};
+
+    tc_flex_list* fl1 = make_flex_list_double(v1);
+    tc_sarray* sa1 = tc_sarray_create_from_list(fl1, &error);
+
+    turi::flex_list lst1;
```

Comment:
It's fine in these test cases to put `using namespace turi;` at the top so you don't have to write `turi::` at the start of each one.